### ISSUE TITLE
Finalisations

### ISSUE THREAD

**Issue by derekfountain** [derekfountain/zxwonkyonekey#15]
It's getting to the point where I have to call it a day. I'd like to move on to something else!

These are the things which need completing:

* finish the final level.
* review the platform graphics and colours. Many are rather horrible.
* a splash screen describing the keys, particularly the sound ones.
* fix the teleporter-in-slowdown bug.
* sound effect of collecting key.
* do something more imaginative with the scoring.

**Comment by derekfountain** [comment]
I think for the splash I can repurpose the ticker. Ticker at the start saying what to do, fireworks at the end for the winner celebration.

**Comment by derekfountain** [comment]
I think for the splash I can repurpose the ticker. Ticker at the start saying what to do, fireworks at the end for the winner celebration.

**Comment by derekfountain** [comment]
Repurposing the ticker to run at the start of the game didn't work. It looks rubbish. I need a splash screen.

See 'splash' branch.

**Comment by derekfountain** [comment]
I hate the colours on level 3 - green on white with a red border, blue/cyan jumpers. Yuk!

Have another go at this one.

**Comment by derekfountain** [comment]
I think the l end of level sound effect would be better used for the teleporter.

**Comment by derekfountain** [comment]
I tweaked the sounds effects for end of level and teleporter. It's an improvement, but there's only so much you can do with that 1 bit sound. :)

**Comment by derekfountain** [comment]
Review of what's left here:

* splash screen
* teleporter-in-slowdown bug
* scoring and bonus

**Comment by derekfountain** [comment]
Splash screen is complete, implemented as an "intro" level. Works well. Scoring and bonus is fixed as well.

I need to:

* fix the teleporter bug
* change functions to fastcall where possible
* take out as many local variables as possible
* fix the music player so it used z88dk best practise
* check statesave for final version
* check it works with the ROM writes disabled
** decide if I'm going to ship with no tracing and set it accordingly
* bump version to 1.00

Most of these are little tasks. Almost there. :)

**Comment by derekfountain** [comment]
I tried changing functions to fastcall, but ZSDCC seems to produce invalid code. I haven't been able to reproduce a testcase, but it does it quite reliably in the main code. I'll come back to this.

I tried taking out the local variables and replacing them with static globals, but it made the code much bigger and a bit slower. I think the conventional wisdom that statics are faster than locals is wrong. I've asked for input on the z88dk list.

**Comment by derekfountain** [comment]
I converted the music player ASM routine to Z88DK fastcall convention. Not that it makes any difference to anything, but at least it's now correct. :)

I looked at the statesave quite thoroughly. The version is in there at the top and rest looks good.

Music is enabled by default now. Dear Lord...

I decided there's no harm shipping with tracing enabled. It doesn't slow the game down, so why not leave it in? It might be useful at some point.
